doc/pandas-2.0/source/goals.rst

```diff
@@ -0,0 +1,194 @@
+.. _goals:
+
+=======================
+ Goals and Motivations
+=======================
+
+.. note::
+
+  These documents are largely written by Wes McKinney, and at this point
+  reflect his opinions for the time being. Many things may change as we discuss
+  and work to reach a consensus about the path forward.
+
+The pandas codebase is now over 8 years old, having grown to over 200,000 lines
+of code from its original ~10,000 LOC in the original 0.1 open source release
+in January 2010.
+
+At a high level, the "pandas 2.0" effort is based on a number of observations:
+
+* The pandas 0.x series of releases have consisted with huge amounts of
+  iterative improvements to the library along with some major new features, bug
+  fixes, and improved documentation. There have also been a series of
+  deprecations, API changes, and other evolutions of pandas's API to account
+  for suboptimal design choices (for example: the ``.ix`` operator) made in the
+  early days of the project (2010 to 2012).
+* The unification of Series and DataFrame internals to be based on a common
+  ``NDFrame`` base class and "block manager" data structure (originally created
+  by me in 2011, and heroically driven forward to its modern form by Jeff
+  Reback), while introducing many benefits to pandas, has come to be viewed as
+  a long-term source of technical debt and code complexity.
+* pandas's ability to support an increasingly broad set of use cases has been
+  significantly constrained (as will be examined in detail in these documents)
+  by its tight coupling to NumPy and therefore subject to various limitations
+  in NumPy.
+* Making significant functional additions (particularly filling gaps in NumPy)
+  to pandas, particularly new data types, has grown increasingly complex with
+  very obvious accumulations of technical debt.
+* pandas is being used increasingly for very large datasets on machines with
+  many cores and large amounts of RAM (100s of gigabytes to terabytes). It
+  would be nice to be able to better utilize these larger, beefier systems
+  within a single Python process.
+* pandas is being used increasingly as a computational building block of some
+  larger system, such as Dask or Apache Spark. We should consider reducing the
+  overhead for making data accessible to pandas (i.e. via memory-mapping or
+  other low-overhead memory sharing).
+* Rough edges in pandas's implementation (e.g. its handling of missing data
+  across data types) are being exposed to users.
+
+These documents are largely concerned with pandas's internal design, which is
+mostly invisible to average users. Advanced users of pandas are generally
+familiar with some of these internal details, particular around performance and
+memory use, and so the degree to which users are impacted will vary quite a
+lot.
+
+Key areas of work
```

Comment:
Agreed -- staying focused on the goals (without diving into the possible solutions) will help steer the discussion. If something is controversial, it helps to return to the goal since tradeoffs / API semantics changes will have to be made in the interest of achieving some of the goals. There may also be other approaches to achieving the results that we haven't thought of.